framework/include/media/MediaRecorder.h

```diff
@@ -36,20 +36,35 @@
 
 namespace media {
 
+class MediaRecorderImpl;
+
 /**
  * @brief result of call the apis
  * @details @b #include <media/MediaRecorder.h>
  * @since TizenRT v2.0 PRE
  */
-typedef enum recorder_result_e {
+typedef enum recorder_result_e : int {
 	/** MediaRecorder Error case */
-	RECORDER_ERROR,
+	RECORDER_ERROR_NOT_ALIVE = -16,
+	RECORDER_ERROR_INVALID_STATE,
+	RECORDER_ERROR_INVALID_FRAME,
+	RECORDER_ERROR_INVALID_VOLUME,
+	RECORDER_ERROR_INVALID_PARAM,
+	RECORDER_ERROR_INVALID_DATASOURCE,
+	RECORDER_ERROR_INIT_AUDIO_FAILED,
+	RECORDER_ERROR_FILE_OPEN_FAILED,
+	RECORDER_ERROR_SET_AUDIO_FAILED,
+	RECORDER_ERROR_RESET_AUDIO_FAILED,
+	RECORDER_ERROR_CAPTURE_AUDIO_FAILED,
+	RECORDER_ERROR_PAUSE_AUDIO_FAILED,
+	RECORDER_ERROR_STOP_AUDIO_FAILED,
+	RECORDER_ERROR_OUT_OF_MEMORY,
+	RECORDER_ERROR_SET_VOLUME_AUDIO_FAILED,
```

Comment:
There will be several operations what using device's functionality, for example bass, treble.
In this case we can use specific value for all errors if device doesn't support such a function.
@chanijjani already added that, but there should be conflict between his pr and this.